File: packages/icoder-sdk/examples/streams-smoke.mjs
import { readFileSync } from 'node:fs';
import iCoDer from '../dist/index.js';

const baseURL = process.env.ICODER_E2E_STREAMS_BASE_URL;
const accessToken = process.env.ICODER_E2E_ACCESS_TOKEN;
const tenantName = process.env.ICODER_E2E_TENANT_NAME;
const audioPath = process.env.ICODER_E2E_STREAMS_AUDIO_PATH;
if (!baseURL || !accessToken || !tenantName || !audioPath) {
  throw new Error('missing local Streams E2E environment');
}
const audio = readFileSync(audioPath);

const interactionId = crypto.randomUUID();
const sdk = new iCoDer({ baseURL, auth: { accessToken } });
const session = await sdk.streams.connect({
  interactionId,
  tenantName,
  environment: 'cn',
  configuration: {
    transcription: {
      primaryLanguage: 'zh-CN', diarize: false, isMultichannel: false,
      participants: [{ channel: 0, role: 'multiple' }],
    },
    mode: { type: 'transcription' },
    retentionPolicy: 'retain',
    audioFormat: 'audio/ogg; codecs=opus',
  },
});
const messages = [];
session.on('message', (message) => messages.push(message));
session.sendAudio(audio);
session.flush();
session.end();
await Promise.race([
  session.waitForEnded(),
  new Promise((_, reject) => setTimeout(() => reject(new Error('Streams timeout')), 15000)),
]);
const types = messages.map((message) => message.type);
for (const required of ['flushed', 'delta_usage', 'usage', 'ENDED']) {
  if (!types.includes(required)) throw new Error(`missing Streams event: ${required}`);
}
const errorCodes = messages
  .filter((message) => message.type === 'error')
  .map((message) => message.code)
  .sort();
if (JSON.stringify(errorCodes) !== JSON.stringify(['STT_UNAVAILABLE'])) {
  throw new Error('local disabled-provider errors were not explicit and bounded');
}
const usage = messages.find((message) => message.type === 'usage');
if (usage?.credits !== 0) throw new Error('local Streams usage must not invent credits');
session.close();
console.log(JSON.stringify({
  sdk: 'javascript', status: 'passed', interaction_id: interactionId,
  retention_policy: 'retain', event_types: types,
  expected_error_codes: errorCodes,
  synthetic_audio_bytes: audio.length, synthetic_silence_ogg_opus: true,
  real_stt_engine_used: false, real_llm_used: false,
}));

File: packages/icoder-python/examples/streams_smoke.py
"""Real loopback Streams smoke using generated silent Ogg/Opus audio."""

from __future__ import annotations

import asyncio
import json
import os
import uuid
from pathlib import Path

from icoder_sdk import iCoDerClient, iCoDerConfig


async def main() -> None:
    base_url = os.environ.get("ICODER_E2E_STREAMS_BASE_URL", "")
    access_token = os.environ.get("ICODER_E2E_ACCESS_TOKEN", "")
    tenant_name = os.environ.get("ICODER_E2E_TENANT_NAME", "")
    audio_path = os.environ.get("ICODER_E2E_STREAMS_AUDIO_PATH", "")
    if not base_url or not access_token or not tenant_name or not audio_path:
        raise RuntimeError("missing local Streams E2E environment")
    audio = Path(audio_path).read_bytes()
    interaction_id = str(uuid.uuid4())
    client = iCoDerClient(iCoDerConfig(base_url=base_url, access_token=access_token))
    messages: list[dict] = []
    try:
        session = await client.streams.connect_async(
            interaction_id=interaction_id,
            tenant_name=tenant_name,
            environment="cn",
            configuration={
                "transcription": {
                    "primaryLanguage": "zh-CN",
                    "diarize": False,
                    "isMultichannel": False,
                    "participants": [{"channel": 0, "role": "multiple"}],
                },
                "mode": {"type": "transcription"},
                "retentionPolicy": "none",
                "audioFormat": "audio/ogg; codecs=opus",
            },
        )
        session.on("message", messages.append)
        await session.send_audio(audio)
        await session.flush()
        await session.end()
        await asyncio.wait_for(session.wait_ended(), timeout=15)
        types = [message.get("type") for message in messages]
        for required in ("flushed", "delta_usage", "usage", "ENDED"):
            if required not in types:
                raise RuntimeError(f"missing Streams event: {required}")
        error_codes = sorted(
            message.get("code")
            for message in messages
            if message.get("type") == "error"
        )
        if error_codes != ["STT_UNAVAILABLE"]:
            raise RuntimeError("local disabled-provider errors were not explicit and bounded")
        usage = next(message for message in messages if message.get("type") == "usage")
        if usage.get("credits") != 0.0:
            raise RuntimeError("local Streams usage must not invent credits")
        await session.close()
        print(json.dumps({
            "sdk": "python",
            "status": "passed",
            "interaction_id": interaction_id,
            "retention_policy": "none",
            "event_types": types,
            "expected_error_codes": error_codes,
            "synthetic_audio_bytes": len(audio),
            "synthetic_silence_ogg_opus": True,
            "real_stt_engine_used": False,
            "real_llm_used": False,
        }, sort_keys=True))
    finally:
        client.close()


if __name__ == "__main__":
    asyncio.run(main())
